Dockerfile

```diff
@@ -44,7 +48,10 @@ COPY api/v2/go.mod api/v2/go.sum   /opt/dex/dependencies/api/v2/
 # Import frontend assets and set the correct CWD directory so the assets
 # are in the default path.
 COPY web web
+COPY examples/config.tmpl /etc/dex/config.tmpl
 
-ENTRYPOINT ["dex"]
+RUN chown -R 1001:1001 /etc/dex/
+USER 1001:1001
 
-CMD ["version"]
+ENTRYPOINT ["dockerize", "-no-overwrite", "-template", "/etc/dex/config.tmpl:/etc/dex/config.yaml"]
+CMD ["dex", "serve", "/etc/dex/config.yaml"]
```

Comment:
What about moving `"dex"` to the end of ENTRYPOINT, to avoid a breaking change for current users of the Docker image?

```
ENTRYPOINT ["dockerize", "-no-overwrite", "-template", "/etc/dex/config.tmpl:/etc/dex/config.yaml", "dex"]
CMD ["serve", "/etc/dex/config.yaml"]
```

Reply:
Seems reasonable. Thanks!

Reply:
Personally, I never like when an application is the entrypoint. It just ties the user's hands. Although it is a breaking change, but I don't think it'll affect most users. We could think of some sort of deprecation for this change to warn users in advance, but I'm not sure how effective it would be.

Reply:
I hardly believe that adding a deprecation warning will help. Mentioning such a change in the RM should be enough. However, for me, it looks better to leave `dex` at the entry point just because changing behavior is not required by this PR.